tests/api_workflow/test_api_workflow_artifacts.py

```diff
@@ -160,3 +161,28 @@ def test__download_compute_worker_run_artifact_by_type__no_artifact_with_type(
             output_path="output_dir/checkpoint.ckpt",
             timeout=0,
         )
+
+
+def test__get_compute_worker_run_checkpoint_url(
+    mocker: MockerFixture,
+) -> None:
+    mocked_client = mocker.MagicMock(spec=ApiWorkflowClient)
```

Comment:
Cannot be a `spec_set`, because we overwrite the `_compute_worker_api`

Reply:
Did you consider not mocking the full client, like this:
```
def test__get_compute_worker_run_checkpoint_url(
    mocker: MockerFixture,
) -> None:
    client = ApiWorkflowClient(token="123")
    client._compute_worker_api.get_docker_run_artifact_read_url_by_id.return_value = (
        "some_read_url"
    )

    mocked_run = mocker.MagicMock(spec_set=DockerRunData)
    read_url = client.get_compute_worker_run_checkpoint_url(run=mocked_run)
    ...
```

This would be easier to understand and it is clearer what exactly has to be mocked.

Reply:
Yes, I considered that option, because it is also used for the other tests, but decided against it:
1. It also tests the `ApiWorkflowClient.__init__`.
2. It calls the version check and thus the API inside `ApiWorkflowClient.__init__`
3.  I know from typing the last worker tests that that option does not work with mypy: It is NOT allowed to mock only some methods of an existing class:

```python
client = ApiWorkflowClient(token="123")
client.some_method = mocker.MagicMock() # raises a mypy assignment error, because you cannot assign a MagicMock() to a Callable.
```